lib/zabbix/zabbix_test.go

```diff
@@ -0,0 +1,97 @@
+package zabbix
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+// Test that we can get Zabbix API key from an API call,
+// set it in API struct
+func TestGetAndSetKey(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintln(w, `{"result": "fake_zabbix_key"}`)
+	}))
+	defer ts.Close()
+
+	var api API
+	api.User = "ZABBIX_USER"
+	api.Password = "ZABBIX_PASSWORD"
+	api.URL = ts.URL
```

Comment:
You could also try your `NewAPI` here and get a free test coverage ;) More importantly, in case it expands in the future to include some side effects